roles/openshift_facts/library/openshift_facts.py

```diff
@@ -534,6 +534,27 @@ def set_aggregate_facts(facts):
 
     return facts
 
+def set_openstack_facts(facts):
+    """ Set openstack facts
+
+        Args:
+            facts (dict): existing facts
+        Returns:
+            dict: the facts dict updated with openstack facts
+    """
+
+    def _add_openstack_config(conf):
+        conf.setdefault('cloud-provider', []).append('openstack')
+        conf.setdefault('cloud-config', []).append('/etc/cloud.conf')
+
+    if 'master' in facts:
+        _add_openstack_config(facts['master'].setdefault('api_server_args', {}))
+        _add_openstack_config(facts['master'].setdefault('controller_args', {}))
+
+    if 'node' in facts:
+        _add_openstack_config(facts['node'].setdefault('kubelet_args', {}))
+
+    return facts
```

Comment:
Thanks for the review. guess_host_provider can indeed be used for this.
I'm not sure to get what you mean by "merge the *_args from the provider info with *_args in fact after the user defined changes are applied". Are you thinking about doing this merge in `openstack_facts.py` ?